### ISSUE TITLE
 dropzone view: targetplatform.windows is not supported 

### ISSUE THREAD

**Issue by Asad0473** [deakjahn/flutter_dropzone#61]
I am using this library in my web app not desktop app but I am still facing this issue . its working fine in debug mode but  when we go to live server or you say release mode  i am facing this issue

**Comment by deakjahn** [comment]
Could you come up with a reproducible repo?

**Comment by Asad0473** [comment]
what do you mean by  reproducible repo?

**Comment by deakjahn** [comment]
A very small project, a minimal one, that shows the problem and allows me to reproduce it. Because, as things stand now, I can't make that error appear myself, so I have no idea what leads to it and how I can fix it.

**Comment by Asad0473** [comment]
@deakjahn , here is the sample project link. please download it and make it live on server. because we are getting target platform error on our live server. in local it's working fine. our app is developed for Android, IOS, and web. but this we allowing this feature is only for web . please make it supported for Window and MacOs as well. I think it might be creating problem because this library is not supporting Windows and MacOs.

https://drive.google.com/file/d/1644K50Sx_-imnKUYcEyqlJTj58xHXkCx/view?usp=sharing

**Comment by Asad0473** [comment]
@deakjahn We are working on a social media website called Werfie (https://werfie.com) which is similar to Twitter. We would love to dropzone developed by you in werfie. You will get credit for it as well in our website Credits section. I request you to please update us with the issue that we are facing while using dropzone in our website. We are expecting to launch the new version of werfie.com on Monday next week and this issue is holding the release. I request your immediate attention on this.
waiting for your response...

**Comment by deakjahn** [comment]
@Asad0473 Then please, help me what to look for :-) I downloaded it, put it into Android Studio and built for the web. I uploaded it to my test site, currenly located at http://teszt.tramontana-teszt.hu/#/. When I start it, it starts, when I drag an image into it, it appears. I'm testing in Vivaldi, a Chrome based browser but I also tried Edge to be sure.

**Comment by Asad0473** [comment]
Its not working on my website

**Comment by deakjahn** [comment]
But does my test at the link above work for you, in your browser?

**Comment by Asad0473** [comment]
@deakjahn  no your test link is not working on my site

**Comment by Asad0473** [comment]
sorry its working fine in my browser

**Comment by Asad0473** [comment]
but its not work for https://werfie.com/ ..

**Comment by Asad0473** [comment]
Maybe its not working for secure website

**Comment by deakjahn** [comment]
I don't think so but I can add a cert to my test in a few minutes. In the meantime, could you do the same as I did? I built your sample simply with:

    flutter build web

or

    flutter build web --web-renderer canvaskit

and upload the contents of `build/web` to anywhere on your site into a new folder that I can access from outside?

**Comment by deakjahn** [comment]
It's HTTPS now: https://teszt.tramontana-teszt.hu/#/

**Comment by Asad0473** [comment]
Ok i test it

**Comment by deakjahn** [comment]
I don't really know where you plan to use it on the site. In the dialog to send a post?

**Comment by Asad0473** [comment]
yes in dialog to post

**Comment by deakjahn** [comment]
Right now I get a window showing the image and the browser has CORS error messages. But nothing about Windows...

**Comment by Asad0473** [comment]
https://staging.werfie.com/ use this link

**Comment by deakjahn** [comment]
I tried with the real one, with a disposable e-mail, but didn't send any message. :-) I'll try to register the same here.

**Comment by Asad0473** [comment]
![tttt](https://user-images.githubusercontent.com/88876294/212170168-d35b3242-f787-42e7-98f9-318020b7b652.PNG)
see this image

**Comment by deakjahn** [comment]
I can't register because I can't select a country, the dropdown doesn't open. It did all right on the real site, although it would be nice to have a way to use the keyboard to locate the country, there are quite a few on the Earth. ;-))

**Comment by Asad0473** [comment]
Can you see the above image ?

**Comment by deakjahn** [comment]
Yes, sure, but I'd like to see it in action so that I can look into the browser dev console.

**Comment by Asad0473** [comment]
Email: Asad.iqbal@softech.com.pk
Password:12345678
Use this credential
On this link https://staging.werfie.com/.. And see what happened in console

**Comment by deakjahn** [comment]
OK, I saved it, you can delete the message.

**Comment by Asad0473** [comment]
Ok

**Comment by deakjahn** [comment]
I can see the message, but I don't yet know what Windows has to do with it. :-) There are Firebase errors in the console but that's something else.

**Comment by Asad0473** [comment]
Yes i know .. I also could not find that problem its working fine in debug mode and even android release mode.. But we go to live window error is coming. I think this libaray is not for windows  thats why

**Comment by deakjahn** [comment]
It certainly isn't, it's a web only plugin, but we're on the web now. :-) Basically, this is a federated plugin. This means that it has a central interface and implementation for platforms. In this particular case, a single implementation, nothing else: web. All the others (Android, iOS, Linux, macOS, Web) are missing because there is no sense, of course. That means that the interface itself throws this error for any implementation: https://github.com/deakjahn/flutter_dropzone/blob/master/flutter_dropzone_platform_interface/lib/method_channel_flutter_dropzone.dart

And then, the web implementation overrides is and does the actual work: https://github.com/deakjahn/flutter_dropzone/blob/master/flutter_dropzone_web/lib/flutter_dropzone_plugin.dart

So, this is the genuine error message that you should see on anything _but_ the web. But even then, how come Flutter thinks it's running on Windows?

**Comment by deakjahn** [comment]
[defaultTargetPlatform](https://api.flutter.dev/flutter/foundation/defaultTargetPlatform.html) comes from the Flutter engine, `package:flutter/foundation.dart`, Flutter fills it with the platform it thinks it runs on. What does it say when you start the program? Could you log it somewhere when you start running?

**Comment by Asad0473** [comment]
I dont see any error on log  my log is clear

**Comment by deakjahn** [comment]
It's not necessarily important:

![Screenshot 2023-01-12 214432](https://user-images.githubusercontent.com/2234271/212177144-6aff6047-d76f-4a73-86f8-c5a69a13d480.png)

**Comment by Asad0473** [comment]
I think this is not related to this

**Comment by deakjahn** [comment]
No, certainly not.

**Comment by Asad0473** [comment]
By the way  i used getmaterial from getx not material app

**Comment by Asad0473** [comment]
You know already do that.. I place it some other location but i got same error

**Comment by deakjahn** [comment]
Not a Dropzone, just a Text to see what defaultTargetPlatform says.

**Comment by Asad0473** [comment]
I do not understand what are you trying to say that

**Comment by deakjahn** [comment]
You have that dialog with the text field to type into, the avatar of the user, the Dropzone, icon buttons at the bottom. I would add a `Text()` widget as well that shows the value of defaultTargetPlatform from foundation. Just for a minute, to see what is in this variable.

**Comment by Asad0473** [comment]
<img width="602" alt="Screenshot 2023-01-13 114104" src="https://user-images.githubusercontent.com/88876294/212254570-39d41725-6b3b-40bd-a85e-4f475f00bf0b.png">
defaultTargetPlatform shows TargetPlatform.windows

**Comment by Asad0473** [comment]
please can you also tell me your flutter version

**Comment by Asad0473** [comment]
flutter build web --web-renderer html --release 
 i run this command and its working fine now
 thanks for your time and effort . I really appreciateit :-)

**Comment by deakjahn** [comment]
OK, that's good to hear, a temporary relief but not the solution, unfortunately, because it should have no connection with that. I only use CanvasKit since it appeared (not even in stable, but already before that) because it provides much better performance and features than the old one. Even the test I put up yesterday was CanvasKit. So, this isn't the real solution, even if it now helps you. :-)

**Comment by JinHwanOh** [comment]
Had same issue.. flutter build web --web-renderer html --release did the trick.

**Comment by srmanc** [comment]
For me it doesn't work and I AM using html as the web renderer. I am testing this on Linux.

**Comment by deakjahn** [comment]
And do you all have the problem in your own, fully developed app, or maybe we can create a minimal one that can be shared here? Because I'm yet to see it and this makes it awfully hard to diagnose... :-))

**Comment by srmanc** [comment]
Switched to drop_zone, sorry.

**Comment by Ahnaf16** [comment]
I'm having the same problem.
here is my code : 
```
Column(
        crossAxisAlignment: CrossAxisAlignment.start,
        children: [
          Text('default Target Platform : $defaultTargetPlatform'),
          const SizedBox(height: 10),
          const Text('is Web : $kIsWeb'),
          const SizedBox(height: 10),
          if (kIsWeb)
            DropzoneView(
              onCreated: (controller) => dropCtrl = controller,
              onDrop: (value) {
                log(value.name);
              },
            ),
        ],
      ),`
but still having `dropzone view: targetplatform.windows is not supported` as output.
![image](https://user-images.githubusercontent.com/55395335/219378795-84249d89-8e92-4ab2-bdf4-2243ef128915.png)
even though I'm on the web, defaultTargetPlatform is detecting windows cause I'm developing on windows?

**Comment by deakjahn** [comment]
@Ahnaf16
> cause I'm developing on windows?

So do I and still I don't have the error. Strange. Yes, this shows Windows for me, too, but the drop still works all right. Anyway, I try to come up with a solution.

**Comment by deakjahn** [comment]
To put it into some perspective, we never actually check the platform, there's no decision based on that. Simply, this is a web plugin that has its implementation in `flutter_dropzone_plugin.dart`. You only get this error message if, for some reason, it doesn't register and start the plugin. As I still can't trigger the error, could you help me a bit with this? This is how it works (or is supposed to work):

We have an abstract class and the static `instance` is initialized with it:

https://github.com/deakjahn/flutter_dropzone/blob/903b16c22151b14c14f40cb818e58b083674ede6/flutter_dropzone_platform_interface/lib/flutter_dropzone_platform_interface.dart#L54

But as soon as the real web plugin is registered, this `instance` gets overwritten:

https://github.com/deakjahn/flutter_dropzone/blob/903b16c22151b14c14f40cb818e58b083674ede6/flutter_dropzone_web/lib/flutter_dropzone_plugin.dart#L20-L25

And from now on, any part the uses this instance will use the web plugin. You only get the error message if this registration doesn't work and the instance keeps the original value (where all functions lead to a very deliberate exception:

https://github.com/deakjahn/flutter_dropzone/blob/903b16c22151b14c14f40cb818e58b083674ede6/flutter_dropzone_platform_interface/lib/method_channel_flutter_dropzone.dart#L14-L15

So, the question would be if you could debug this `registerWith()` function, maybe with some log readouts, to see how long you can get in your code.

**Comment by Ahnaf16** [comment]
> To put it into some perspective, we never actually check the platform, there's no decision based on that. Simply, this is a web plugin that has its implementation in `flutter_dropzone_plugin.dart`. You only get this error message if, for some reason, it doesn't register and start the plugin. As I still can't trigger the error, could you help me a bit with this? This is how it works (or is supposed to work):
> 
> We have an abstract class and the static `instance` is initialized with it:
> 
> https://github.com/deakjahn/flutter_dropzone/blob/903b16c22151b14c14f40cb818e58b083674ede6/flutter_dropzone_platform_interface/lib/flutter_dropzone_platform_interface.dart#L54
> 
> But as soon as the real web plugin is registered, this `instance` gets overwritten:
> 
> https://github.com/deakjahn/flutter_dropzone/blob/903b16c22151b14c14f40cb818e58b083674ede6/flutter_dropzone_web/lib/flutter_dropzone_plugin.dart#L20-L25
> 
> And from now on, any part the uses this instance will use the web plugin. You only get the error message if this registration doesn't work and the instance keeps the original value (where all functions lead to a very deliberate exception):
> 
> https://github.com/deakjahn/flutter_dropzone/blob/903b16c22151b14c14f40cb818e58b083674ede6/flutter_dropzone_platform_interface/lib/method_channel_flutter_dropzone.dart#L14-L15
> 
> So, the question would be if you could debug this `registerWith()` function, maybe with some log readouts, to see how far you can get in your code.

I don't know what happened, I was away for a day. But it's working now ...  I think the real web plugin failed to registered for some reason.

**Comment by deakjahn** [comment]
Yes, this is what [federated plugins](https://docs.flutter.dev/development/packages-and-plugins/developing-packages#federated-plugins) are about, all current Flutter plugins that support various platforms are expected to use this structure.

Yes, I removed that file. Flutter moved from Android embedding v1 to v2 gradually, less and less plugins and apps used the old one, so it no longer gets generated these days. I don't have it, either, but it remained in the repo. Even when it was used, it was generated automatically.

I'm glad to hear it works for you but then we didn't solve the problem this time around, either. :-) Anyway, I can only ask anybody who still experiences this issue to put a breakpoint or a log write into that `registerWith()` to see whether it gets called and executed properly.

**Comment by Attila-Wryte** [comment]
<img width="487" alt="Screenshot 2023-04-05 at 16 09 22" src="https://user-images.githubusercontent.com/97024647/230108415-c1fa37b8-5b45-49da-96b4-8c60521ca843.png">
I had the same issue with macOS. Flutter clean solved it for me.

**Comment by deakjahn** [comment]
Then please, read the comment just above and try to apply a breakpoint or a log write to see.

**Comment by MenuItem207** [comment]
Had a similar issue that I resolved by running 'flutter clean'

**Comment by crizant** [comment]
I encountered it once, also on MacOS. Didn't tweak anything, just built the flutter web once again and the error is gone.

**Comment by zeevgrim** [comment]
> Facing the same problem with mac-OS.
> 
> I know the plugin is for the web, and we are also building a website. When we deploy our website, such an error occurs of the target platform is unsupported.

same, only after deployment.

**Comment by Ahnaf16** [comment]
for me, setting the web renderer to canvaskit when building for release solved my problem. But still don't know what exactly caused the problem.

**Comment by toco1001** [comment]
`flutter build web --web-renderer html --release` works for me.

**Comment by congthanhng** [comment]
i got same, but 
`flutter build web --web-renderer html --release` NOT works for me.

**Comment by deakjahn** [comment]
May I reiterate my message from Feb 18? We won't be able to find it out without somebody actually seeing this problem looking into the breakpoint mentioned there...

**Comment by Rimzone** [comment]
> Had a similar issue that I resolved by running 'flutter clean'

NB

**Comment by huseyinseyit** [comment]
DropzoneView: TargetPlatform.windows is not supported.

-My project only web project not desktop or android, ios.
-There is no error log.
-Working localhost but hosting firebase and real devices, browsers don't work.

**Comment by deakjahn** [comment]
Please, have a look at https://github.com/deakjahn/flutter_dropzone/issues/61#issuecomment-1433883055

**Comment by DavidOrakpo** [comment]
I'm running into this same issue as well

**Comment by DavidOrakpo** [comment]
I'm not sure how to go about the 61 comment you referenced @deakjahn. Can you break it down? 
My case is exactly as has been explained above. I'm using canvas kit

**Comment by deakjahn** [comment]
@DavidOrakpo As you probably found in that comment, the problem is not directly in the plugin but in Flutter erroneously calling it when it shouldn't. As I never was able to reproduce it, I can't really determine anything until somebody whole actually experiences the issue can debug it.

This shouldn't be that hard, all browsers have very large and sophisticated Developer Tools (Ctrl+Shift+I or F12). You could find the code mentioned above and put a breakpoint, much like in any other programming environment.

**Comment by deakjahn** [comment]
We're not yet sure whether https://github.com/deakjahn/flutter_dropzone/pull/89 solves this. Could those of you who have problems with this try that modification?

**Comment by mrcsh** [comment]
I am also seeing this issue with my app.  I think I  have tracked it down to the fact that none of the plugin registration code is being emitted in main.dart.js so I  it is just simply using the skeleton code in the base stub implementation.

I built the profile version of the example and my app and looked at the resulting main.dart.js files.   The example has many references to FlutterDropzonePlugin* functions.  My app has zero, just references to the transpiled FlutterDropzonePlatform code from the stub dart implementation.   

My usage of flutter_dropzone is in a different package than the one with the main.dart for the app.  We build several flavors of the app and each has it's only small main that does some setup and then calls the core main app code in a different package.  I tried adding the flutter_dropzone  dependency to the stub (main) package  but that did not change anything.

**Comment by mrcsh** [comment]
So it's a bit of a hack, but I got it to work by referencing the web plugin directly in my app's main.
I had to create a dummy plugin class so the non-web builds would compile, but that seemed to be enough to get the plugin registered in main.dart.js  

That doesn't address why it's getting missed in the first place.

**Comment by deakjahn** [comment]
The problem is still the old one: I never happened to be able to reproduce it myself, and also it very much looks like it happens outside the plugin, in the Flutter code that initializes (or fails to initialize) it.

**Comment by SAGV** [comment]
For anyone having the issue, I actually managed to get rid of it by… adding 

```
  plugin:
    platforms:
      web:
        default_package: flutter_dropzone
```

to pubspec. 

However, to confirm the fix, I've removed the following lines and the whole thing still… works. 

So I'm not really sure what this did. Maybe by editing pubspec file it refreshed some caches?

**Comment by imanabu** [comment]
> For anyone having the issue, I actually managed to get rid of it by… adding
> 
> ```
>   plugin:
>     platforms:
>       web:
>         default_package: flutter_dropzone
> ```
> 
> to pubspec.
> 
> However, to confirm the fix, I've removed the following lines and the whole thing still… works.
> 

This did not work in my situation.  I also performed `fluter clean` to make sure I grab all the pubs from scratch.

**Comment by imanabu** [comment]
> So it's a bit of a hack, but I got it to work by referencing the web plugin directly in my app's main. I had to create a dummy plugin class so the non-web builds would compile, but that seemed to be enough to get the plugin registered in main.dart.js
> 
> That doesn't address why it's getting missed in the first place.

- Referencing it in `main.dart` definitely was the fix for my situation.

This is what I did. In main.dart, I added DropzoneView(); in InitState()

```
class TheAppState extends ConsumerState<TheApp> {
  // This widget is the root of your application.

  @override
  void initState() {
    super.initState();
    // This is a hack that will emit the Dropzone json file during the
    // production build. 
    DropzoneView();
   ... your other stuff...
  }
```

Also, I have two Flutter client projects and one I did before worked perfectly without any of this. The latest project, I had this issue.

**Comment by bjterry99** [comment]
> > So it's a bit of a hack, but I got it to work by referencing the web plugin directly in my app's main. I had to create a dummy plugin class so the non-web builds would compile, but that seemed to be enough to get the plugin registered in main.dart.js
> > That doesn't address why it's getting missed in the first place.
> 
> * Referencing it in `main.dart` definitely was the fix for my situation.
> 
> This is what I did. In main.dart, I added DropzoneView(); in InitState()
> 
> ```
> class TheAppState extends ConsumerState<TheApp> {
>   // This widget is the root of your application.
> 
>   @override
>   void initState() {
>     super.initState();
>     // This is a hack that will emit the Dropzone json file during the
>     // production build. 
>     DropzoneView();
>    ... your other stuff...
>   }
> ```
> 
> Also, I have two Flutter client projects and one I did before worked perfectly without any of this. The latest project, I had this issue.

This did not resolve the issue for me.
Still no solution to this problem?

**Comment by bjterry99** [comment]
Solution
https://github.com/deakjahn/flutter_dropzone/issues/72#issuecomment-2385579151
